base/dict.jl

```diff
@@ -87,14 +87,13 @@ mutable struct Dict{K,V} <: AbstractDict{K,V}
 
     function Dict{K,V}() where V where K
         n = 16
-        new(zeros(UInt8,n), Vector{K}(undef, n), Vector{V}(undef, n), 0, 0, 0, 1, 0)
+        new(fill(0x80,n), Vector{Pair{K,V}}(undef, n), 0, 0, 0, 1, 0)
```

Comment:
Would it make sense to have 0x00 represent empty, so we can start with zero'd memory in case that ever becomes especially fast or convenient?

Reply:
Probably good idea, but it should have no effect for now. I'll try that. The coding of metadata would be:
``` julia
# Metadata: empty => 0x00, removed => 0x7f, full => 0b1[7 most significant hash bits]
```

Reply:
Updated and there is no significant performance difference, only a single extra `|` operation.
https://github.com/JuliaLang/julia/blob/8fd9617f7906ec78cde6d867e43857a64aa56eff/base/dict.jl#L170-L172